### PR TITLE
spotify-qt: init at v3.5

### LINKED ISSUE #114785: package request: spotify-qt - "Lightweight Spotify client using Qt"
**Project description**

> A Spotify client using Qt as a simpler, lighter alternative to the official client, inspired by spotify-tui. Much like spotify-tui, you need an actual Spotify client running, for example spotifyd, which can be configured from within the app. Also like other clients, controlling music playback requires Spotify Premium.

**Metadata**

* homepage URL: https://github.com/kraxarn/spotify-qt
* source URL: https://github.com/kraxarn/spotify-qt
* license: gpl3
* platforms: unix/linux

### PR DESCRIPTION
**Pull request by karthikiyengar**
Resolves #114785

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/spotify-qt/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  cmake,
+  mkDerivation,
+  libxcb,
+  qtbase,
+  qtsvg
+}:
+
+mkDerivation rec {
+   pname = "spotify-qt";
+   version = "v3.5";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify-qt/default.nix:14]:
```suggestion
   version = "3.5";
```

File: pkgs/applications/audio/spotify-qt/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+   src = fetchFromGitHub {
+      owner = "kraxarn";
+      repo = pname;
+      rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify-qt/default.nix:19]:
```suggestion
      rev = "v${version}";
```

File: pkgs/applications/audio/spotify-qt/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  fetchFromGitHub,
+  wrapQtAppsHook,
+  lib,
+  cmake,
+  mkDerivation,
+  libxcb,
+  qtbase,
+  qtsvg
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify-qt/default.nix:10]:
```suggestion
{ fetchFromGitHub
, wrapQtAppsHook
, lib
, cmake
, mkDerivation
, libxcb
, qtbase
, qtsvg
}:
```

File: pkgs/applications/audio/spotify-qt/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+   nativeBuildInputs = [ cmake ];
+
+   cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=" ];
+
+   installFlags = [ "DESTDIR=$(out)" ];
+
+   meta = with lib; {
+    description = "Lightweight unofficial Spotify client using Qt";
+    homepage = "https://github.com/kraxarn/spotify-qt";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify-qt/default.nix:34]: gpl3Only or gpl3Plus?

**Review by SuperSandro2000 — COMMENTED** [review]
other than that LGTM

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118261` at c2d9f8ea run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>spotify-qt</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `wrapQtAppsHook`.
Near pkgs/applications/audio/spotify-qt/default.nix:2:3:
```
  |
2 | , wrapQtAppsHook
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/audio/spotify-qt/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+   cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=" ];
+
+   installFlags = [ "DESTDIR=$(out)" ];
+
+   meta = with lib; {
+    description = "Lightweight unofficial Spotify client using Qt";
+    homepage = "https://github.com/kraxarn/spotify-qt";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ kiyengar ];
+    platforms = with platforms; unix;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify-qt/default.nix:34]:
```suggestion
    platforms = platforms.unix;
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by colemickens** [comment]
Cool, thanks!

**Pull request merged**